logs-proxy/proxy.yaml

```diff
@@ -0,0 +1,226 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: logs-proxy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      infra: logs-proxy
+  template:
+    metadata:
+      name: logs-proxy
+      labels:
+        infra: logs-proxy
+    spec:
+      containers:
+      - name: app
+        image: docker.io/library/python:3-alpine
+        command: ["python3", "/app/s3-proxy.py", "https://cockpit-logs.us-east-1.linodeobjects.com", "10080"]
+        volumeMounts:
+        - name: logs-proxy-app
+          mountPath: /app/
+          readOnly: true
+        - name: s3-credentials
+          mountPath: /s3
+          readOnly: true
+
+      - name: anubis
+        image: ghcr.io/techarohq/anubis:latest
+        ports:
+          - containerPort: 8080
+            protocol: TCP
+            name: anubis-port
+        env:
+          # https://anubis.techaro.lol/docs/admin/installation/
+          - name: BIND
+            value: ":8080"
+          - name: METRICS_BIND
+            value: ":9099"
+          - name: SERVE_ROBOTS_TXT
+            value: "true"
+          - name: TARGET
+            # app container listens on port 10080
+            value: "http://localhost:10080/"
+          - name: DIFFICULTY
+            value: "6"
+          - name: COOKIE_EXPIRATION_TIME
+            value: "24h"
+
+      volumes:
+      - name: logs-proxy-app
+        configMap:
+          name: logs-proxy-app
+          defaultMode: 0755
+      - name: s3-credentials
+        secret:
+          secretName: cockpit-s3-log-read-secrets
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: logs-proxy-app
+data:
+  s3-proxy.py: |
```

Comment:
Can this become a separate file which we include in the future, that would allow us to lint the code

Reply:
I find it a bit awkward as well, but it's hard with kubernetes -- you can't have bare files, you have to ship them as part of a ConfigMap or a container image. The latter feels too much overhead for the first experiment. 

We *could* split it out and instead of a static yaml file create a shell script that constructs the yaml on the fly for deployment. I'll look at that.